zookeeper-server/src/main/java/org/apache/zookeeper/server/NodeHashMapImpl.java

```diff
@@ -33,6 +33,7 @@
     private final ConcurrentHashMap<String, DataNode> nodes = new ConcurrentHashMap<String, DataNode>();
 
     private AdHash hash = new AdHash();
+    private final DigestCalculator digestCalculator = new DigestCalculator();
```

Comment:
Rather than instantiating new object here, request for it in the constructor. You should get the same object as `DataTree` is using.

Reply:
you are right, thanks! I just pushed this change.